google/auth/oauth2client.py

```diff
@@ -0,0 +1,133 @@
+        info)
+
+
+def _convert_gce_app_assertion_credentials(credentials):
+    """Converts to :class:`google.auth.compute_engine.Credentials`."""
+    return google.auth.compute_engine.Credentials(
+        service_account_email=credentials.service_account_email)
+
+
+def _convert_appengine_app_assertion_credentials(credentials):
+    """Converts to :class:`google.auth.app_engine.Credentials`."""
+    return google.auth.app_engine.Credentials(
+        scopes=(_helpers.string_to_scopes(credentials.scope)
+                if credentials.scope is not None else None),
+        service_account_id=credentials.service_account_id)
+
+
+_CLASS_CONVERSION_MAP = {}
+
+if _HAS_OAUTH2CLIENT:
+    _CLASS_CONVERSION_MAP.update({
+        oauth2client.client.OAuth2Credentials: _convert_oauth2_credentials,
+        oauth2client.client.GoogleCredentials: _convert_oauth2_credentials,
+        oauth2client.service_account.ServiceAccountCredentials:
+            _convert_service_account_credentials,
+        oauth2client.service_account._JWTAccessCredentials:
+            _convert_service_account_credentials,
+        oauth2client.contrib.gce.AppAssertionCredentials:
+            _convert_gce_app_assertion_credentials,
+    })
+
+if _HAS_APPENGINE:
+    _CLASS_CONVERSION_MAP[
+        oauth2client.contrib.appengine.AppAssertionCredentials] = (
+            _convert_appengine_app_assertion_credentials)
+
+
+def convert(credentials):
+    """Convert oauth2client credentials to google-auth credentials.
+
+    This class converts:
+
+    - :class:`oauth2client.client.OAuth2Credentials` to
+      :class:`google.oauth2.credentials.Credentials`.
+    - :class:`oauth2client.client.GoogleCredentials` to
+      :class:`google.oauth2.credentials.Credentials`.
+    - :class:`oauth2client.service_account.ServiceAccountCredentials` to
+      :class:`google.oauth2.service_account.Credentials`.
+    - :class:`oauth2client.service_account._JWTAccessCredentials` to
+      :class:`google.oauth2.service_account.Credentials`.
+    - :class:`oauth2client.contrib.gce.AppAssertionCredentials` to
+      :class:`google.auth.compute_engine.Credentials`.
+    - :class:`oauth2client.contrib.appengine.AppAssertionCredentials` to
+      :class:`google.auth.app_engine.Credentials`.
+
+    Returns:
+        google.auth.credentials.Credentials: The converted credentials.
+
+    Raises:
+        EnvironmentError: If oauth2client is not installed.
+        ValueError: If the credentials could not be converted.
+    """
+    if not _HAS_OAUTH2CLIENT:
+        raise EnvironmentError(
+            'oauth2client is not installed and is required in order to convert'
+            'credentials.')
+
+    credentials_class = type(credentials)
+
+    try:
+        return _CLASS_CONVERSION_MAP[credentials_class](credentials)
+    except KeyError:
+        raise ValueError(
+            'Unable to convert {} to a google-auth credentials class.'.format(
```

Comment:
I don't like putting the error string so far away from where I throw, but fine, done.